### ISSUE TITLE
Regex parser should have the option to 'skip' some number of header rows like the CSV parser can

### ISSUE THREAD

**Issue by vogievetsky** [apache/incubator-druid#8583]
This would be super useful for ingesting data that has some form of a header such as what is seen in https://github.com/apache/incubator-druid/issues/8555.

**Comment by tan31989** [comment]
Hi,

I would like to work on this PR.

**Comment by vogievetsky** [comment]
Awesome!

**Comment by vogievetsky** [comment]
@tan31989 are you still interested in working on this?

**Comment by tan31989** [comment]
@vogievetsky yes, I'm trying to figure out the required changes. Kind of stuck with figuring out the linked issue with this

**Comment by tan31989** [comment]
@vogievetsky I have tried X number of ways, trying to copy the CSVParser kind of implementations. Pardon me if this is vague, but I see the following code is using: `if (!matcher.matches()) {}` in [here](https://github.com/apache/incubator-druid/blob/master/core/src/main/java/org/apache/druid/java/util/common/parsers/RegexParser.java#L93) is used for matching entire text.

I feel that beats the purpose of Regex parser, where if the pattern does not match until the entire text is matched as a whole. I was of the opinion it would best fit the use cases, where we use: `while (matcher.find()) {}`, thus providing us with the ability to write regex with more flexibilities.

With `matcher.find()` it's easier to replicate a regex pattern find and group. Adding a regex to match an entire string as is always ends up with using a global filter like `(.*)`. There are so many variants of regex that would be missed because of this.

**Comment by yashwanth-Thota** [comment]
I would like to work on this
